apps/api/src/app/auth/usecases/switch-organization/switch-organization.usecase.ts

```diff
@@ -23,7 +24,12 @@ export class SwitchOrganization {
 
     const member = await this.memberRepository.findMemberByUserId(command.newOrganizationId, command.userId);
     const user = await this.userRepository.findById(command.userId);
-    const token = await this.authService.getSignedToken(user, command.newOrganizationId, member);
+    const environment = await this.environmentRepository.findOne({
+      _organizationId: command.newOrganizationId,
+      _parentId: { $exists: false },
+    });
+
+    const token = await this.authService.getSignedToken(user, command.newOrganizationId, member, environment._id);
```

Comment:
```suggestion
    const token = await this.authService.getSignedToken(user, command.newOrganizationId, member, environment?._id);
```

Perhaps in some usecase the environment is undefined and it failing because of it?